public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts

```diff
@@ -31,6 +31,40 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
     this.resourcePath = `${routeNames.resourceGraph}`;
   }
 
+  async fetchInitialRows(currentSelection?: string): Promise<ResourceRowGroup> {
```

Comment:
copying a comment from the other pr: a more readable interface for this function would be `fetchInitialRows(subscription?: string, resourceGroup?: string)`. That way it's more clear that depending on what you pass you will get a different set of resources.

Reply:
I'm not sure that this suggestion completely makes sense to me in this context?

For example if you called `fetchInitialRows` with a subscription and nothing else it would work the same as if you passed it nothing, which I would think is more confusing. 

As you have written it there would be no way to pass it a specific resource either?

The way this function works is it will:
- always fetch all the subscriptions
- then if there is a current selection, it will load the resource groups within the selected subscription
- and if the current selection is a resource it will also load all the resources within the selected resource group. 

Does that make sense?

Reply:
mmh, yeah I misread the function here but does it make sense?

I know it works the way it is at the moment but it's kind of weird that you fetch resource groups only if you set a resource group (even if the resource group is not a parameter for that fetch). The same for the resource name.

It kind of makes sense to me to refactor this and say:
 - If you set nothing return all subscriptions.
 - If you set a subscription, return the resource groups in that 
 - If you set a resource group, return the resources in that group

(the resource name is not used at all)

You would need a separate piece of logic (another function?) that given a uri, returns what you need to fetch:
 - If the uri points to a resource, fetch resources, groups and subs
 - if the uri points to a group, fetch subs and groups
 - if the uri points to a subs, fetch only subs

That way you have two functions with only one responsibility. A [pedantic link](https://en.wikipedia.org/wiki/Single-responsibility_principle) :smile:.